### PR TITLE
Add base ruff config + lint top level files

### PR DESCRIPTION
**Pull request by nateprewitt**
This PR is the one in a series of steps to move the AWS CLI codebase onto a standard set of formatting (`ruff`) between all related tools (boto3, botocore, s3transer, awscli and awscli v2). This PR builds on the recent changes to remove six from the codebase and starts by adding our intended ruff configurations and running them over all files matching `awscli/*.py`. This does not include customizations, tests, or our scripts.

The plan currently is to have 3-4 smaller PRs to break up the cognitive overhead for reviewers. We'll tackle the primary files in this PR, then customizations and scripts. Once we're confident the existing test suite is still working with all CLI code changed, we'll migrate the test suite as well. The last piece will be to turn on our linting GHA (boto3 example) and have this standardized across all future PRs.

### CODE REVIEW

File: awscli/alias.py [hunk 1]
```diff
@@ -126,13 +131,17 @@ def name(self, value):
...
 
 class ServiceAliasCommand(BaseAliasCommand):
-    UNSUPPORTED_GLOBAL_PARAMETERS = [
-        'debug',
-        'profile'
-    ]
-
-    def __init__(self, alias_name, alias_value, session, command_table,
-                 parser, shadow_proxy_command=None):
+    UNSUPPORTED_GLOBAL_PARAMETERS = ('debug', 'profile')
```
Comment (kdaily) [inline awscli/alias.py:134]: Is the type change here from `list` to `tuple` OK?

File: awscli/compat.py [hunk 2]
```diff
@@ -6,33 +6,30 @@
...
 
-from urllib.error import URLError
+from botocore.compat import six, OrderedDict
 
-from botocore.compat import six
-from botocore.compat import OrderedDict
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
```
Comment (jonathan343) [inline awscli/compat.py:29]: Looks like the copyright message was affected here

File: awscli/completer.py [hunk 3]
```diff
@@ -5,28 +5,29 @@
...
 #     http://aws.amazon.com/apache2.0/
 
+import copy
+import logging
+import sys
+
 # or in the "license" file accompanying this file. This file is
 # distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
```
Comment (jonathan343) [inline awscli/completer.py:15]: Same copyright issue here.

File: awscli/handlers.py [hunk 4]
```diff
@@ -106,23 +132,25 @@ def awscli_initialize(event_handlers):
...
-                            add_examples)
+    #    # The following will get fired for every option we are
+    #    # documenting.  It will attempt to add an example_fn on to
+    #    # the parameter object if the parameter supports shorthand
+    #    # syntax.  The documentation event handlers will then use
+    #    # the examplefn to generate the sample shorthand syntax
+    #    # in the docs.  Registering here should ensure that this
+    #    # handler gets called first but it still feels a bit brittle.
+    #    event_handlers.register('doc-option-example.*.*.*',
+    #                            param_shorthand.add_example_fn)
```
Comment (jonathan343) [inline awscli/handlers.py:143]: Not relevant in the scope of the PR, but this comment block should probably be removed. It's been here for 10 years.

File: awscli/shorthand.py [hunk 5]
```diff
@@ -348,27 +339,30 @@ def _expect(self, char, consume_whitespace=False):
...
 
     def _must_consume_regex(self, regex):
-        result = regex.match(self._input_value[self._index:])
+        result = regex.match(self._input_value[self._index :])
         if result is not None:
             return self._consume_matched_regex(result)
-        raise ShorthandParseSyntaxError(self._input_value, '<%s>' % regex.name,
-                                        '<none>', self._index)
+        raise ShorthandParseSyntaxError(
+            self._input_value, f'<regex.name>', '<none>', self._index
```
Comment (jonathan343) [inline awscli/shorthand.py:359]: this seems wrong. I would thing this is updated to `f'<{regex.name}>'` not `f'<regex.name>'`

File: awscli/table.py [hunk 6]
```diff
@@ -6,19 +6,19 @@
 
 #     http://aws.amazon.com/apache2.0/
 
+import struct
+
 # or in the "license" file accompanying this file. This file is
 # distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
```
Comment (jonathan343) [inline awscli/table.py:14]: another copyright issue

File: awscli/table.py [hunk 7]
```diff
@@ -167,25 +181,39 @@ def __init__(self):
     def style_title(self, text):
         # Originally bold + underline
         return text
-        #return colorama.Style.BOLD + text + colorama.Style.RESET_ALL
+        # return colorama.Style.BOLD + text + colorama.Style.RESET_ALL
```
Comment (jonathan343) [inline awscli/table.py:184]: another comment we can probably delete

**Review by jonathan343 — COMMENTED** [review]
Left a few comments. We should also update the contributing guidance with instructions on how to install and run pre-commit.

**Comment by jonathan343** [comment]
We should also consider adding a lint workflow similar to botocore:
https://github.com/boto/botocore/blob/develop/.github/workflows/lint.yml

File: awscli/alias.py (hunk 1 above)
Comment (nateprewitt) [reply awscli/alias.py:134]: Yeah, that should be fine. This should have been a tuple to begin with.

**Comment by nateprewitt** [comment]
> We should also consider adding a lint workflow similar to botocore:
https://github.com/boto/botocore/blob/develop/.github/workflows/lint.yml

I covered that in the overview, this is one of 4 PRs so we can't run ruff right now till they're all done. The last step listed is to add the GHA.

> We should also update the contributing guidance with instructions on how to install and run pre-commit.

You can't run `pre-commit` successfully until all the PRs are merged, so we'll omit the guidance for now so it doesn't create confusion. It'll be added with the GHA above.

Other comments are good catches, I'll get those addressed tomorrow.

**Comment by jonathan343** [comment]
> > We should also consider adding a lint workflow similar to botocore:
> > https://github.com/boto/botocore/blob/develop/.github/workflows/lint.yml
> 
> I covered that in the overview, this is one of 4 PRs so we can't run ruff right now till they're all done. The last step listed is to add the GHA.
> 
> > We should also update the contributing guidance with instructions on how to install and run pre-commit.
> 
> You can't run `pre-commit` successfully until all the PRs are merged, so we'll omit the guidance for now so it doesn't create confusion. It'll be added with the GHA above.
> 
> Other comments are good catches, I'll get those addressed tomorrow.

My bad, missed that. This PR LGTM now, approving!

**Review by jonathan343 — APPROVED** [review]

**Pull request merged**